`test_messages_where_eth_account_matches_metamask_but_not_ethers` failed

### What happened?

I'm packaging for Arch Linux User Repository and Arch Linux CN. Some tests are failing.

```text
$ npm version
{
  npm: '11.0.0',
  node: '23.4.0',
  acorn: '8.14.0',
  ada: '2.9.2',
  amaro: '0.2.0',
  ares: '1.34.3',
  brotli: '1.1.0',
  cjs_module_lexer: '1.4.1',
  cldr: '45.0',
  icu: '75.1',
  llhttp: '9.2.1',
  modules: '131',
  napi: '9',
  nbytes: '0.1.1',
  ncrypto: '0.0.1',
  nghttp2: '1.64.0',
  openssl: '3.4.0',
  simdjson: '3.10.1',
  simdutf: '5.6.3',
  sqlite: '3.47.1',
  tz: '2024a',
  undici: '6.21.0',
  unicode: '15.1',
  uv: '1.49.2',
  uvwasi: '0.0.21',
  v8: '12.9.202.28-node.11',
  zlib: '1.3.1'
}
```

### Code that produced the error

```bash
# Maintainer: Xeonacid <h.dwwwwww@gmail.com>

_name=eth-account
pkgname=python-${_name}
pkgver=0.13.4
pkgrel=1
pkgdesc="Account abstraction library for web3.py"
arch=(any)
url="https://github.com/ethereum/${_name}"
license=(MIT)
depends=(python python-bitarray python-eth-abi python-eth-keyfile python-eth-keys python-eth-rlp python-eth-utils python-hexbytes python-rlp python-ckzg python-pydantic)
makedepends=(python-build python-installer python-setuptools python-wheel python-sphinx python-sphinx_rtd_theme)
checkdepends=(python-pytest python-hypothesis nodejs npm)
source=(${_name}-${pkgver}.tar.gz::${url}/archive/v${pkgver}.tar.gz)
sha512sums=('ed9d39fcb4a8f15fde73494be369f568c11c922f5344f9246de60d71ed5001b685558f46be85a2f5bfb29ea1c2acc0b1c5141f2377ab92e7759e490d15eb6bd1')

prepare() {
  cd $_name-$pkgver
  cd tests/integration/js-scripts
  npm ci
}

build() {
  cd $_name-$pkgver
  python -m build --wheel --no-isolation
  make -C docs man
}

check(){
  cd $_name-$pkgver
  python -m venv --system-site-packages test-env
  test-env/bin/python -m installer dist/*.whl
  test-env/bin/python -m pytest -vv --showlocals tests/
}

package() {
  cd $_name-$pkgver
  python -m installer --destdir="$pkgdir" dist/*.whl
  install -Dm644 docs/_build/man/eth_account.1 -t "$pkgdir/usr/share/man/man1"
  install -Dm644 LICENSE -t "$pkgdir/usr/share/licenses/$pkgname"
  install -Dm644 README.md -t "$pkgdir/usr/share/doc/$pkgname"
}
```


### Full error output

```shell
=========================== short test summary info ============================
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_space_in_custom_type_fieldname] - AssertionError: assert '0x5269ead91219d629d6b5e7b1deef3482b672c9a96c50d2df6c420fe638f4a78e525627bb77196661272e9111e22cf57bcb9fe0ed34f67ae85b651cd75202d4651c' == '0x093277f1d6a54dd40f298881163deabd4e45876fd65da528e925ec3f0d9f7cbc5a10fed0e00ec2b706daeac7f3ff23e9f8ecbbcd5138322d89f09d0c6680df641c'
  
  - 0x093277f1d6a54dd40f298881163deabd4e45876fd65da528e925ec3f0d9f7cbc5a10fed0e00ec2b706daeac7f3ff23e9f8ecbbcd5138322d89f09d0c6680df641c
  + 0x5269ead91219d629d6b5e7b1deef3482b672c9a96c50d2df6c420fe638f4a78e525627bb77196661272e9111e22cf57bcb9fe0ed34f67ae85b651cd75202d4651c
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_multi_no_fixed_len_array] - AssertionError: assert '0x543c4fadbb4559f1ef6c6fb9d4b05e33144d17b313b5c612fca5f8cb1f585fe957dc9c9c81f35860b856716b795f84e5818693a806d0502d729693a2923a33501b' == '0x1fcdbdbf01c5d65dde754891cb23ec182f489e57662573492216bd3025f69c5a4aab307b52876aafdf80a82fd01e58da21bf7b3d7950f46999401501afe392a51b'
  
  - 0x1fcdbdbf01c5d65dde754891cb23ec182f489e57662573492216bd3025f69c5a4aab307b52876aafdf80a82fd01e58da21bf7b3d7950f46999401501afe392a51b
  + 0x543c4fadbb4559f1ef6c6fb9d4b05e33144d17b313b5c612fca5f8cb1f585fe957dc9c9c81f35860b856716b795f84e5818693a806d0502d729693a2923a33501b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_eip712_example] - AssertionError: assert '0x33600224dbc5a598b7c443b0cc9241b4b12ea10441244cc058442b31065b37232258c5d0b55f22362b1fa97d5c7fca4f40bfc1545417d974a22dfd4a8ac2de8b1b' == '0x1ffea3c07dff27ba4da68583b9a91c2784633aa7498a8dc87e3759531d7cac6a741c5b09f3648369af273965e59698b67a8cdd3d140027bae42bba0dda48c15d1b'
  
  - 0x1ffea3c07dff27ba4da68583b9a91c2784633aa7498a8dc87e3759531d7cac6a741c5b09f3648369af273965e59698b67a8cdd3d140027bae42bba0dda48c15d1b
  + 0x33600224dbc5a598b7c443b0cc9241b4b12ea10441244cc058442b31065b37232258c5d0b55f22362b1fa97d5c7fca4f40bfc1545417d974a22dfd4a8ac2de8b1b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_issue_201_with_array] - AssertionError: assert '0xe2a156ae136785d74c63cb72c3121954854e31e43d69c92dae1afbd845ad9c5209189f71ffdc3eec6cab82509d7bdc0cd43d2016825b40487e61dbf4e580eb881b' == '0xf334ce746ac897bf3d79db563d220305474275bd5b103478612169b5aa23786e2ab7c6e08e0f58425a5a9b76dde819f9f8109e0efb8a462aeece5796ffd8dfa31c'
  
  - 0xf334ce746ac897bf3d79db563d220305474275bd5b103478612169b5aa23786e2ab7c6e08e0f58425a5a9b76dde819f9f8109e0efb8a462aeece5796ffd8dfa31c
  + 0xe2a156ae136785d74c63cb72c3121954854e31e43d69c92dae1afbd845ad9c5209189f71ffdc3eec6cab82509d7bdc0cd43d2016825b40487e61dbf4e580eb881b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_eip712_example_with_array] - AssertionError: assert '0xfcd28bfd425a26215d645481af446895f761b16896ac007c86433136ad9bb5c815ccdef9a6947396be54e92955c5c707ce65099d0371b4488143264675f5e18b1c' == '0xed2a0daeb90d33ebce90159ada202a4272d35b0ff6a587485a550f768ea12da24ed2c3895c184f546864f580c4d4c19a36a7b2531942351481de88412b8be2f01c'
  
  - 0xed2a0daeb90d33ebce90159ada202a4272d35b0ff6a587485a550f768ea12da24ed2c3895c184f546864f580c4d4c19a36a7b2531942351481de88412b8be2f01c
  + 0xfcd28bfd425a26215d645481af446895f761b16896ac007c86433136ad9bb5c815ccdef9a6947396be54e92955c5c707ce65099d0371b4488143264675f5e18b1c
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_eip712_example_addresses_not_checksummed] - AssertionError: assert '0x33600224dbc5a598b7c443b0cc9241b4b12ea10441244cc058442b31065b37232258c5d0b55f22362b1fa97d5c7fca4f40bfc1545417d974a22dfd4a8ac2de8b1b' == '0x1ffea3c07dff27ba4da68583b9a91c2784633aa7498a8dc87e3759531d7cac6a741c5b09f3648369af273965e59698b67a8cdd3d140027bae42bba0dda48c15d1b'
  
  - 0x1ffea3c07dff27ba4da68583b9a91c2784633aa7498a8dc87e3759531d7cac6a741c5b09f3648369af273965e59698b67a8cdd3d140027bae42bba0dda48c15d1b
  + 0x33600224dbc5a598b7c443b0cc9241b4b12ea10441244cc058442b31065b37232258c5d0b55f22362b1fa97d5c7fca4f40bfc1545417d974a22dfd4a8ac2de8b1b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_issue_201_no_array] - AssertionError: assert '0xf21e900f15ac36898c347e2f68a2655ad785c64a49b983a0f864ec14f62b1c3753c6d0c011d2c092a9fe664258795449f1460f935fe57047839d9495a7e610611b' == '0xa06c5baa95240451c9bcd1b04c5e5e4d4b02463fa94de4bf5731f35aef56393f26ec132300f8ef6a4ff63fc49411bcb81b432bd86cca591ed73243b5f8ada7841b'
  
  - 0xa06c5baa95240451c9bcd1b04c5e5e4d4b02463fa94de4bf5731f35aef56393f26ec132300f8ef6a4ff63fc49411bcb81b432bd86cca591ed73243b5f8ada7841b
  + 0xf21e900f15ac36898c347e2f68a2655ad785c64a49b983a0f864ec14f62b1c3753c6d0c011d2c092a9fe664258795449f1460f935fe57047839d9495a7e610611b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_multi_array] - AssertionError: assert '0xc9c325fc7ae95d2df72be62dfa884b73eb040cf26078e7e0b45d5c58f8ba0a404f2b381e71252999d46b4d817f8977b82ccca8fcfc9c98736d2b9f6c18c68bcb1b' == '0xe3c3decb23dfe13d889907b33443177f3424c88ea772620ef0260952969b4858453d13d01274d35ea009918d9902c51e252e6a905c07a083293667c775c8f1901b'
  
  - 0xe3c3decb23dfe13d889907b33443177f3424c88ea772620ef0260952969b4858453d13d01274d35ea009918d9902c51e252e6a905c07a083293667c775c8f1901b
  + 0xc9c325fc7ae95d2df72be62dfa884b73eb040cf26078e7e0b45d5c58f8ba0a404f2b381e71252999d46b4d817f8977b82ccca8fcfc9c98736d2b9f6c18c68bcb1b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_all_3_sigs_match[valid_primary_type_is_empty_string] - AssertionError: assert '0xb488c6735d72bbcbd5ea0865e00836c717d7b60b9c21cae38fe47a1fc409b9b245dba55a02cdf51af8177eb83904d33ca7c5adb93edac305506318002c705d7c1c' == '0x87854d08176d0d73a26a137db5c9ed83a92f6eaab0ea0965676396dc8698c1d65f7d4244d6bd0d8c5efcb8f14a0afc509aa57ffb03acbee32a1f3be1a0e1ee021b'
  
  - 0x87854d08176d0d73a26a137db5c9ed83a92f6eaab0ea0965676396dc8698c1d65f7d4244d6bd0d8c5efcb8f14a0afc509aa57ffb03acbee32a1f3be1a0e1ee021b
  + 0xb488c6735d72bbcbd5ea0865e00836c717d7b60b9c21cae38fe47a1fc409b9b245dba55a02cdf51af8177eb83904d33ca7c5adb93edac305506318002c705d7c1c
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_eth_account_matches_metamask_but_not_ethers[valid_int_value_for_string_type_ethers_handles_mismatched_type_and_value_differently] - AssertionError: assert '0xf65299465004ea38d1786c29754c8d19e633668347d9a38db9827e24de95db0105148a78cea33cc6a565c2ae6d0b5436b87bda661c23dabe28a89de97b38fc4c1c' == '0x15a5a29005a25dc70a7376cc76643dc9cb1a4e70b7d1f094586d64a64ec9152402b96ce4903de59878f65d3141ac661d066d31b5954110c23e55007ba1a33ef41b'
  
  - 0x15a5a29005a25dc70a7376cc76643dc9cb1a4e70b7d1f094586d64a64ec9152402b96ce4903de59878f65d3141ac661d066d31b5954110c23e55007ba1a33ef41b
  + 0xf65299465004ea38d1786c29754c8d19e633668347d9a38db9827e24de95db0105148a78cea33cc6a565c2ae6d0b5436b87bda661c23dabe28a89de97b38fc4c1c
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_eth_account_matches_metamask_but_not_ethers[valid_null_value_for_custom_type_ethers_requires_non_null] - AssertionError: assert '0x882913820aaa24f4fd08f7e099ee72dc90faf325bbc9ce452d03fd27bd5cdef83b4fa03a8d9ec1e6a7c89b2cc59b0fe282b05ab2f8cf35aac72cef90780627701c' == '0x40534443024f6af083b64c1bc50128e1485099fb62ffa0715ea94f595b02cab44a441ff5cc17f7cdba1b8422da72e9f2e0166ddedefe788ec52d25f140ce371b1c'
  
  - 0x40534443024f6af083b64c1bc50128e1485099fb62ffa0715ea94f595b02cab44a441ff5cc17f7cdba1b8422da72e9f2e0166ddedefe788ec52d25f140ce371b1c
  + 0x882913820aaa24f4fd08f7e099ee72dc90faf325bbc9ce452d03fd27bd5cdef83b4fa03a8d9ec1e6a7c89b2cc59b0fe282b05ab2f8cf35aac72cef90780627701c
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_eth_account_matches_metamask_but_not_ethers[valid_nested_structs_ethers_does_not_accept_recursive_custom_type_arrays] - AssertionError: assert '0xac42d967e741d771a391f3bbe0f9e1759a8f1f06fa057ea0f46285c88f834c8632077ec17a8b7ac972b6decda73719ccaef645f28422afa24cdcbde7cdf465b01b' == '0xa08754ba82844b47ede3d3b162350cef111e08279efb16b521e1f233ec7fb98a6101e3493ac53a00c994231e9f7f542c67c921795b338f121f3619cabc0f59251b'
  
  - 0xa08754ba82844b47ede3d3b162350cef111e08279efb16b521e1f233ec7fb98a6101e3493ac53a00c994231e9f7f542c67c921795b338f121f3619cabc0f59251b
  + 0xac42d967e741d771a391f3bbe0f9e1759a8f1f06fa057ea0f46285c88f834c8632077ec17a8b7ac972b6decda73719ccaef645f28422afa24cdcbde7cdf465b01b
FAILED tests/integration/test_comparison_js_eip712_signing.py::test_messages_where_eth_account_matches_metamask_but_not_ethers[valid_eip712_example_ethers_does_not_accept_recursive_custom_types] - AssertionError: assert '0xbf1018700b6469ef64556f9a685e3e70ad7a982f63f6a27c282948d9963c333d678fad80433f2c98e6fef7d1c463f61bd85abffcd569c1ba45f9d48d8875947b1b' == '0x8ff01181b806b2a6620db0fe8c81bce5a010db51e8b94e456b3db39bb0e6184b248a68d6be9f98f89f906c5f7f7738027312463bcbb669b7c346a303684ef3ba1c'
  
  - 0x8ff01181b806b2a6620db0fe8c81bce5a010db51e8b94e456b3db39bb0e6184b248a68d6be9f98f89f906c5f7f7738027312463bcbb669b7c346a303684ef3ba1c
  + 0xbf1018700b6469ef64556f9a685e3e70ad7a982f63f6a27c282948d9963c333d678fad80433f2c98e6fef7d1c463f61bd85abffcd569c1ba45f9d48d8875947b1b
```


### Fill this section in if you know how this could or should be fixed

_No response_

### eth-account Version

0.13.4

### Python Version

3.13.1

### Operating System

linux

### Output from `pip freeze`

```shell
alabaster==1.0.0
annotated-types==0.7.0
attrs==23.2.1.dev0
autocommand==2.2.2
Babel==2.15.0
bitarray==3.0.0
build==1.2.2
cffi==1.17.1
charset-normalizer==3.4.1
ckzg==2.0.1
cryptography==44.0.0
cytoolz==1.0.0
docutils==0.21.2
eth-acount==0.13.4
eth-hash==0.7.0
eth-keyfile==0.8.1
eth-keys==0.6.0
eth-rlp==2.1.0
eth-typing==5.1.0
eth-utils==5.1.0
eth_abi==5.1.0
factory_boy==3.3.0
Faker==33.1.0
filelock==3.16.1
hexbytes==1.2.1
hypothesis==6.123.13
idna==3.10
imagesize==1.4.1
iniconfig==2.0.0
installer==0.7.0
jaraco.collections==5.1.0
jaraco.context==6.0.1
jaraco.functools==4.1.0
jaraco.text==4.0.0
Jinja2==3.1.5
MarkupSafe==2.1.5
more-itertools==10.5.0
packaging==24.2
parsimonious==0.10.0
platformdirs==4.3.6
pluggy==1.5.0
pycparser==2.22
pycryptodome==3.21.0
pydantic==2.10.4
pydantic_core==2.27.2
Pygments==2.19.1
pyproject_hooks==1.2.0
pytest==8.3.4
python-dateutil==2.9.0
pytz==2024.2
regex==2024.11.6
requests==2.32.3
rlp==4.0.1
setuptools==75.2.0
six==1.17.0
snowballstemmer==2.2.0
sortedcontainers==2.4.0
Sphinx==8.1.3
sphinx_rtd_theme==2.0.0
sphinxcontrib-applehelp==2.0.0
sphinxcontrib-devhelp==2.0.0
sphinxcontrib-htmlhelp==2.1.0
sphinxcontrib-jquery==4.1
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==2.0.0
sphinxcontrib-serializinghtml==2.0.0
toolz==1.0.0
typing_extensions==4.12.2
urllib3==2.3.0
wheel==0.45.0
```

--- a/newsfragments/308.internal.rst
+++ b/newsfragments/308.internal.rst
@@ -1,0 +1,1 @@
+``deepcopy`` EIP712 messages used in test to avoid unintended mutation.

--- a/tests/core/test_encode_typed_data/test_encode_typed_data_full_messages.py
+++ b/tests/core/test_encode_typed_data/test_encode_typed_data_full_messages.py
@@ -1,4 +1,7 @@
 import pytest
+from copy import (
+    deepcopy,
+)
 import json
 
 from eth_utils import (
@@ -15,33 +18,35 @@ from tests.eip712_messages import (
     convert_to_3_arg,
 )
 
+all_valid = deepcopy(ALL_VALID_EIP712_MESSAGES)
+invalid = deepcopy(INVALID)
+one_arg_invalid = deepcopy(ONE_ARG_INVALID)
 
-@pytest.mark.parametrize(
-    "test_cases", (ALL_VALID_EIP712_MESSAGES, INVALID, ONE_ARG_INVALID)
-)
+
+@pytest.mark.parametrize("test_cases", (all_valid, invalid, one_arg_invalid))
 def test_there_are_no_duplicate_test_cases(test_cases):
     string_test_cases = [json.dumps(msg, sort_keys=True) for msg in test_cases.values()]
     assert len(string_test_cases) == len(set(string_test_cases))
 
 
-@pytest.mark.parametrize("message", ALL_VALID_EIP712_MESSAGES)
+@pytest.mark.parametrize("message", all_valid)
 def test_valid_messages(message):
-    assert encode_typed_data(
-        full_message=ALL_VALID_EIP712_MESSAGES[message]
-    ) == encode_typed_data(*convert_to_3_arg(ALL_VALID_EIP712_MESSAGES[message]))
+    assert encode_typed_data(full_message=all_valid[message]) == encode_typed_data(
+        *convert_to_3_arg(all_valid[message])
+    )
 
 
-@pytest.mark.parametrize("message", INVALID)
+@pytest.mark.parametrize("message", invalid)
 def test_invalid_messages(message):
     with pytest.raises(ValueError):
-        encode_typed_data(full_message=INVALID[message])
+        encode_typed_data(full_message=invalid[message])
 
     with pytest.raises(ValueError):
-        encode_typed_data(*convert_to_3_arg(INVALID[message]))
+        encode_typed_data(*convert_to_3_arg(invalid[message]))
 
 
-@pytest.mark.parametrize("message", ONE_ARG_INVALID)
+@pytest.mark.parametrize("message", one_arg_invalid)
 def test_messages_that_are_only_invalid_for_one_arg_encoding(message):
     with pytest.raises(ValidationError):
-        encode_typed_data(full_message=ONE_ARG_INVALID[message])
-    encode_typed_data(*convert_to_3_arg(ONE_ARG_INVALID[message]))
+        encode_typed_data(full_message=one_arg_invalid[message])
+    encode_typed_data(*convert_to_3_arg(one_arg_invalid[message]))

--- a/tests/integration/test_comparison_js_eip712_signing.py
+++ b/tests/integration/test_comparison_js_eip712_signing.py
@@ -1,4 +1,7 @@
 import pytest
+from copy import (
+    deepcopy,
+)
 import json
 import subprocess
 
@@ -14,6 +17,10 @@ from tests.eip712_messages import (
     VALID_FOR_PY_AND_METAMASK,
     convert_to_3_arg,
 )
+
+valid_for_all = deepcopy(VALID_FOR_ALL)
+valid_for_py_and_ethers = deepcopy(VALID_FOR_PY_AND_ETHERS)
+valid_for_py_and_metamask = deepcopy(VALID_FOR_PY_AND_METAMASK)
 
 TEST_KEY = "756e69636f726e73756e69636f726e73756e69636f726e73756e69636f726e73"
 py_account = Account.from_key(TEST_KEY)
@@ -34,9 +41,9 @@ def get_js_sig(message, library):
 
 
 @pytest.mark.compatibility
-@pytest.mark.parametrize("message_title", VALID_FOR_ALL)
+@pytest.mark.parametrize("message_title", valid_for_all)
 def test_messages_where_all_3_sigs_match(message_title):
-    message = VALID_FOR_ALL[message_title]
+    message = valid_for_all[message_title]
 
     ethers_sig = get_js_sig(message, "ethers")
     metamask_sig = get_js_sig(message, "metamask")
@@ -59,9 +66,9 @@ def test_messages_where_all_3_sigs_match(message_title):
 
 
 @pytest.mark.compatibility
-@pytest.mark.parametrize("message_title", VALID_FOR_PY_AND_ETHERS)
+@pytest.mark.parametrize("message_title", valid_for_py_and_ethers)
 def test_messages_where_eth_account_matches_ethers_but_not_metamask(message_title):
-    message = VALID_FOR_PY_AND_ETHERS[message_title]
+    message = valid_for_py_and_ethers[message_title]
 
     ethers_sig = get_js_sig(message, "ethers")
     metamask_sig = get_js_sig(message, "metamask")
@@ -85,9 +92,9 @@ def test_messages_where_eth_account_matches_ethers_but_not_metamask(message_titl
 
 
 @pytest.mark.compatibility
-@pytest.mark.parametrize("message_title", VALID_FOR_PY_AND_METAMASK)
+@pytest.mark.parametrize("message_title", valid_for_py_and_metamask)
 def test_messages_where_eth_account_matches_metamask_but_not_ethers(message_title):
-    message = VALID_FOR_PY_AND_METAMASK[message_title]
+    message = valid_for_py_and_metamask[message_title]
 
     ethers_sig = get_js_sig(message, "ethers")
     metamask_sig = get_js_sig(message, "metamask")

